include/dts-subscription.sh

```diff
@@ -0,0 +1,341 @@
+    fi
+    if [ -v HEADS_LINK_DPP ]; then
+      _check_dwn_req_resp_heads=$(curl -L -I -s -f -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$HEADS_LINK_DPP" -o /dev/null -w "%{http_code}")
+    fi
+
+    _check_logs_req_resp=$(curl -L -I -s -f -H "$CLOUD_REQUEST" "$TEST_LOGS_URL" -o /dev/null -w "%{http_code}")
+
+    # Return 0 if any of Dasharo Firmware binaries is available:
+    if [ ${_check_dwn_req_resp_uefi} -eq 200 ] || [ ${_check_dwn_req_resp_heads} -eq 200 ]; then
+      if [ ${_check_logs_req_resp} -eq 200 ]; then
+        print_ok "A Dasharo Firmware binary has been found for your platform!"
+        return 0
+      fi
+    fi
+  fi
+
+  print_warning "Something may be wrong with the DPP credentials or you may not\n
+		have access to Dasharo Firmware. If so, consider getting Dasharo\n
+		Subscription and improving security of your platform!"
+
+  read -p "Press ENTER to continue"
+  return 1
+}
+
+get_dpp_creds() {
+  echo ""
+  read -p "Enter logs key:                " 'TMP_CLOUDSEND_LOGS_URL'
+  echo ""
+  read -p "Enter firmware download key:   " 'TMP_CLOUDSEND_DOWNLOAD_URL'
+  echo ""
+  read -p "Enter password:                " 'TMP_CLOUDSEND_PASSWORD'
+
+  echo ${TMP_CLOUDSEND_LOGS_URL} > ${SE_credential_file}
+  echo ${TMP_CLOUDSEND_DOWNLOAD_URL} >> ${SE_credential_file}
+  echo ${TMP_CLOUDSEND_PASSWORD} >> ${SE_credential_file}
+
+  print_ok "Dasharo DPP credentials have been saved"
+}
+
+login_to_dpp_server(){
+  # Check if the user is already logged in, log in if not:
+  if [ -z "$(mc alias list | grep ${CLOUDSEND_DOWNLOAD_URL})" ]; then
+    mc alias set $DPP_SERVER_USER_ALIAS $DPP_SERVER_ADDRESS $CLOUDSEND_DOWNLOAD_URL $CLOUDSEND_PASSWORD || return 1
+    if [ $? -ne 0 ]; then
+      print_error "Error while logging in to DPP server!"
+      return 1
+    fi
+  fi
+
+  return 0
+}
+
+subscription_routine(){
+  export CLOUDSEND_LOGS_URL
+  export CLOUDSEND_DOWNLOAD_URL
+  export CLOUDSEND_PASSWORD
+
+  # Each time the main menu is rendered, check for DPP credentials and export
+  # them, if file exists
+  if [ -e "${SE_credential_file}" ]; then
+    CLOUDSEND_LOGS_URL=$(sed -n '1p' < ${SE_credential_file} | tr -d '\n')
+    CLOUDSEND_DOWNLOAD_URL=$(sed -n '2p' < ${SE_credential_file} | tr -d '\n')
+    CLOUDSEND_PASSWORD=$(sed -n '3p' < ${SE_credential_file} | tr -d '\n')
+    export USER_DETAILS="$CLOUDSEND_DOWNLOAD_URL:$CLOUDSEND_PASSWORD"
+    export DPP_IS_LOGGED="true"
+  else
+    CLOUDSEND_LOGS_URL="$BASE_CLOUDSEND_LOGS_URL"
+    CLOUDSEND_PASSWORD="$BASE_CLOUDSEND_PASSWORD"
+    unset CLOUDSEND_DOWNLOAD_URL
+    unset DPP_IS_LOGGED
+    return 1
+  fi
+
+  # Network connection may not be available on boot, do not connect if so:
+  check_network_connection && login_to_dpp_server || return 0
+
+  return 0
+}
+
+check_dasharo_package_env(){
+  [ -d $DPP_PACKAGE_MANAGER_DIR ] || mkdir $DPP_PACKAGE_MANAGER_DIR
```

Comment:
```suggestion
  [ -d $DPP_PACKAGE_MANAGER_DIR ] || mkdir -p $DPP_PACKAGE_MANAGER_DIR
```

Reply:
`-p` can create the full structure if needed, I guess it's safer and doesn't hurt to use it

Reply:
Fixed here 6e2ba4b9caaefe8e7cf12a9247479486818efcde and here d80465ed568b6bff1c3e5fafba20146d8e7928d5.